.github/workflows/docs.yaml

```diff
@@ -5,15 +5,26 @@ on:
     paths:
       - 'docs/**'
 
+
 jobs:
 
   build-docs:
     runs-on: ubuntu-latest
+    environment:
+      name: docs
 
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/pyproject.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
```

Comment:
we're caching install, which should (i think) allow subsequent runs to use the cache